astronomer/providers/amazon/aws/hooks/s3.py

```diff
@@ -148,13 +153,20 @@ async def _check_key(
         :param bucket_val: the name of the bucket
         :param key: S3 keys that will point to the file
         :param wildcard_match: the path to the key
+        :param use_regex: whether to use regex to check bucket
         """
+        pattern = key
```

Comment:
No need to reassign the variable. Put it directly in `re.match(pattern=key, string=k["Key"])`.